src/openvic-simulation/modifier/ModifierManager.cpp

```diff
@@ -35,466 +56,517 @@ bool ModifierManager::add_modifier_effect(
 		"REGULAR_EXP_TECH"
 	);
-	ret &= add_modifier_effect(
-		modifier_effect_cache.reinforce_rate, "reinforce_rate", true, PROPORTION_DECIMAL, COUNTRY, "REINFORCE_TECH"
+	ret &= register_technology_modifier_effect(
+		modifier_effect_cache.reinforce_rate, "reinforce_rate", true, PROPORTION_DECIMAL, "REINFORCE_TECH"
 	);
-	ret &= add_modifier_effect(
+	ret &= register_technology_modifier_effect(
 		modifier_effect_cache.separatism, "seperatism", // paradox typo
-		false, PROPORTION_DECIMAL, COUNTRY, "SEPARATISM_TECH"
+		false, PROPORTION_DECIMAL, "SEPARATISM_TECH"
 	);
-	ret &= add_modifier_effect(
-		modifier_effect_cache.shared_prestige, "shared_prestige", true, RAW_DECIMAL, COUNTRY, "SHARED_PRESTIGE_TECH"
+	ret &= register_technology_modifier_effect(
+		modifier_effect_cache.shared_prestige, "shared_prestige", true, RAW_DECIMAL, "SHARED_PRESTIGE_TECH"
 	);
-	ret &= add_modifier_effect(modifier_effect_cache.tax_eff, "tax_eff", true, PERCENTAGE_DECIMAL, COUNTRY, "TECH_TAX_EFF");
+	ret &= register_technology_modifier_effect(modifier_effect_cache.tax_eff, "tax_eff", true, PERCENTAGE_DECIMAL, "TECH_TAX_EFF");
 
 	/* Country Modifier Effects */
-	ret &= add_modifier_effect(
-		modifier_effect_cache.administrative_efficiency, "administrative_efficiency", true, PROPORTION_DECIMAL, COUNTRY
+	ret &= register_technology_modifier_effect(
+		modifier_effect_cache.administrative_efficiency, "administrative_efficiency", true, PROPORTION_DECIMAL
 	);
-	ret &= add_modifier_effect(
+	ret &= register_base_country_modifier_effect(
 		modifier_effect_cache.administrative_efficiency_modifier, "administrative_efficiency_modifier", true,
-		PROPORTION_DECIMAL, COUNTRY, ModifierEffect::make_default_modifier_effect_localisation_key("administrative_efficiency")
-	);
-	ret &= add_modifier_effect(modifier_effect_cache.artisan_input, "artisan_input", false, PROPORTION_DECIMAL, COUNTRY);
-	ret &= add_modifier_effect(modifier_effect_cache.artisan_output, "artisan_output", true, PROPORTION_DECIMAL, COUNTRY);
-	ret &= add_modifier_effect(
-		modifier_effect_cache.artisan_throughput, "artisan_throughput", true, PROPORTION_DECIMAL, COUNTRY
+		PROPORTION_DECIMAL, ModifierEffect::make_default_modifier_effect_localisation_key("administrative_efficiency")
 	);
-	ret &= add_modifier_effect(modifier_effect_cache.badboy, "badboy", false, RAW_DECIMAL, COUNTRY);
-	ret &= add_modifier_effect(
-		modifier_effect_cache.cb_generation_speed_modifier, "cb_generation_speed_modifier", true, PROPORTION_DECIMAL, COUNTRY
+	ret &= register_technology_modifier_effect(modifier_effect_cache.artisan_input, "artisan_input", false, PROPORTION_DECIMAL,{},has_no_effect);
+	ret &= register_technology_modifier_effect(modifier_effect_cache.artisan_output, "artisan_output", true, PROPORTION_DECIMAL,{},has_no_effect);
+	ret &= register_technology_modifier_effect(	modifier_effect_cache.artisan_throughput, "artisan_throughput", true, PROPORTION_DECIMAL,{},has_no_effect);
```

Comment:
```suggestion
	ret &= register_technology_modifier_effect(modifier_effect_cache.artisan_throughput, "artisan_throughput", true, PROPORTION_DECIMAL, {}, has_no_effect);
```
Throughout this function: remove stray tabs, add missing spaces, and split any lines `>= 128` column widths wide.

Reply:
Tabs before EOL removed.
How do you get the IDE to split automatically?

Reply:
You could make it auto-format the file but that would change a lot of other stuff too (including bugging out when formatting the lambda functions and making a massive mess).